nodeJS/orms/prisma_orm.md

```diff
@@ -0,0 +1,157 @@
+
+### Introducing Prisma ORM
+
+ORMs pretty much solve all of the pain points mentioned above but it's not all sunshine and roses. There can be a learning curve to fully understand the ins and outs of an ORM and some don't even fully support all SQL features. Even with these shortcomings, using an ORM can be extremely worthwhile.
+
+In the Node.js ecosystem, there's A LOT of ORMs to choose from, and the community has not landed on a go-to yet. We've decided to cover [Prisma](https://www.prisma.io/) in the curriculum because of its popularity and community support.
+
+Prisma has all the features we need to complete this curriculum and then some. Prisma is just a library which you will `npm install` and use. Let's discuss some of its features. Then will link to some official Prisma guides to get you started.
+
+### Prisma Schema
+
+The Prisma schema is a file where you will define your models. For example, consider a message table in a chat app:
+
+```text
+model Message {
+   id        Int      @id @default(autoincrement())
+   content   String   @db.VarChar(255) 
+   createdAt DateTime @default(now())
+   author    User     @relation(fields: [authorId], references: [id])
+   authorId  Int     
+}
+
+model User {
+   // user's fields
+}
+```
+
+There's a bunch of new stuff up here. Prisma schema files are written in Prisma Schema Language. You can see how not only there are column definitions for the table, but also a relation to another table defined within the `Message` model. You'll learn more about this in the assignment readings.
+
+This schema file lives in your codebase and is tracked by version control. You can probably guess by now, just how useful this is.
+
+### Prisma client
+
+The client is a separate library that you will use to interact with your database. The Prisma client is a bit special in that it's customized to your schema. Um... what?
+
+```javascript
+// instantiate the client
+import { PrismaClient } from '@prisma/client';
+const prisma = new PrismaClient();
+
+// when creating a new message
+await prisma.message.create({
+   data: {
+      content: 'Hello, world!',
+      authorId: 1
+   }
+})
+
+// when fetching all messages
+const messages = await prisma.message.findMany();
+```
+
+Notice `prisma.message` object? How did Prisma know that there's a `message` model? Once you create or update the schema file, all you have to do is run
+
+```bash
+npx prisma generate
+```
+
+in the cli, and Prisma will generate the client for you. The client can handle all sorts of querying: joins, filters, sorting, pagination, and more.
+
+You might have a complex query that you just are unable to get right via the Prisma client, or you feel more comfortable in writing a query raw, you can do that too. Prisma supports raw queries as well.
```

Comment:
```suggestion
You might have a complex query that you just are unable to get right via the Prisma client, or if you feel more comfortable writing a query raw, you can do that too. Prisma supports raw queries as well.
```

Also, I feel like `Prisma supports raw queries as well` is redundant with the previous sentence (being able to write raw queries implies that Prisma supports this), but not sure